assets/src/components/mapPage/vehiclePropertiesCard.tsx

```diff
@@ -121,7 +123,14 @@ const CurrentLocation = ({ vehicle }: VehicleProp): React.ReactElement => {
     vehicle.latitude,
     vehicle.longitude
   )
-  return <>{intersection ?? "Exact location cannot be determined"}</>
+
+  if (isLoading(intersection) && !isOk(intersection)) {
```

Comment:
Hmm that's an interesting idea. I just implemented helper functions that were needed as type guards, and I'm not sure if I could write that in a way that corresponds nicely to a type guard because of how TypeScript's object types permit you to have extra keys beyond what is specified in the type. I think it may become useful as we use these types in more places but I'd rather wait and see how we end up using this and what sorts of helpers end up being useful.